java.util.NoSuchElementException: head of empty list

```
> rho-examples[ERROR] java.util.NoSuchElementException: head of empty list
rho-examples[ERROR] 	at scala.collection.immutable.Nil$.head(List.scala:420)
rho-examples[ERROR] 	at scala.collection.immutable.Nil$.head(List.scala:417)
rho-examples[ERROR] 	at org.http4s.rho.swagger.SwaggerModelsBuilder.mkCollectionProperty$1(SwaggerModelsBuilder.scala:244)
rho-examples[ERROR] 	at org.http4s.rho.swagger.SwaggerModelsBuilder.org$http4s$rho$swagger$SwaggerModelsBuilder$$typeToProp$1(SwaggerModelsBuilder.scala:229)
rho-examples[ERROR] 	at org.http4s.rho.swagger.SwaggerModelsBuilder$$anonfun$mkResponse$1.apply(SwaggerModelsBuilder.scala:256)
rho-examples[ERROR] 	at org.http4s.rho.swagger.SwaggerModelsBuilder$$anonfun$mkResponse$1.apply(SwaggerModelsBuilder.scala:256)
rho-examples[ERROR] 	at scala.Option.map(Option.scala:146)
...
```

--- a/.gitignore
+++ b/.gitignore
@@ -2,6 +2,7 @@
 target/
 /.lib/
 
+*.swp
 *~
 /.ensime
 /.idea/

--- a/examples/src/main/scala/com/http4s/rho/swagger/demo/MyService.scala
+++ b/examples/src/main/scala/com/http4s/rho/swagger/demo/MyService.scala
@@ -13,6 +13,7 @@ import org.http4s.util.UrlFormCodec
 import scalaz._
 import scalaz.Scalaz._
 import scalaz.concurrent.Task
+import scalaz.stream.Process
 
 object MyService extends RhoService with SwaggerSupport {
   import org.http4s.rho._
@@ -87,5 +88,12 @@ object MyService extends RhoService with SwaggerSupport {
   "This route allows your to post stuff" **
     POST / "post" ^ EntityDecoder.text |>> { body: String =>
       "You posted: " + body
+    }
+
+  "This demonstrates using a process of entities" **
+    GET / "stream" |>> {
+      val s = 0 until 100 map (i => s"Hello $i\n")
+      val p: Process[Task, String] = Process.emitAll(s)
+      Ok(p)
     }
 }

--- a/swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala
+++ b/swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala
@@ -8,12 +8,16 @@ import org.http4s.rho.bits.QueryAST.{QueryCapture, QueryRule}
 import org.http4s.rho.bits.ResponseGenerator.EmptyRe
 import org.http4s.rho.bits._
 
+import org.log4s.getLogger
+
 import scala.reflect.runtime.universe._
 
 import scalaz._, Scalaz._
 
 private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
   import models._
+
+  private[this] val logger = getLogger
 
   def mkSwagger(info: Info, ra: RhoAction[_, _])(s: Swagger): Swagger =
     Swagger(
@@ -222,13 +226,19 @@ private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
 
   def mkResponse(code: String, descr: String, otpe: Option[Type]): (String, Response) = {
 
-    def typeToProp(tpe: Type): Property =
-      if (tpe.isPrimitive)
-        mkPrimitiveProperty(tpe)
+    def typeToProp(tpe: Type): Option[Property] =
+      if (Reflector.isExcluded(tpe))
+        None
+      else if (tpe.isPrimitive)
+        mkPrimitiveProperty(tpe).some
       else if (tpe.isCollection)
         mkCollectionProperty(tpe)
+      else if (tpe.isProcess)
+        typeToProp(tpe.dealias.typeArgs(1))
+      else if (tpe.isTask)
+        typeToProp(tpe.dealias.typeArgs(0))
       else
-        RefProperty(ref = tpe.simpleName)      
+        RefProperty(ref = tpe.simpleName).some      
 
     def mkPrimitiveProperty(tpe: Type): Property = {
       import TypeBuilder._
@@ -240,20 +250,27 @@ private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
       }
     }
 
-    def mkCollectionProperty(tpe: Type): Property = {
+    def mkCollectionProperty(tpe: Type): Option[Property] = {
       val param = tpe.dealias.typeArgs.head
       val prop =
         if (param.isPrimitive)
-          mkPrimitiveProperty(param)
+          mkPrimitiveProperty(param).some
         else if (param.isCollection)
           typeToProp(param)
         else
-          RefProperty(ref = param.simpleName)        
+          RefProperty(ref = param.simpleName).some 
 
-      ArrayProperty(items = prop)
+      prop.map(p => ArrayProperty(items = p))
     }
 
-    code -> Response(description = descr, schema = otpe.map(typeToProp))
+    val schema = {
+      try otpe.flatMap(typeToProp)
+      catch { case _: Throwable =>
+        logger.warn(s"Failed to build model for type ${otpe.get}")
+        None
+      }
+    }
+    code -> Response(description = descr, schema = schema)
   }
 
   def mkQueryParam(rule: QueryCapture[_]): QueryParameter =

--- a/swagger/src/main/scala/org/http4s/rho/swagger/package.scala
+++ b/swagger/src/main/scala/org/http4s/rho/swagger/package.scala
@@ -36,7 +36,13 @@ package object swagger {
         typeOf[java.math.BigDecimal], typeOf[java.math.BigInteger])
     }
 
+    private[swagger] val excludes = {
+      Set[Type](typeOf[scala.xml.Elem])
+    }
+
     def isPrimitive(t: Type, extra: Set[Type] = Set.empty) = (primitives ++ extra).exists(t =:= _)
+
+    def isExcluded(t: Type, extra: Set[Type] = Set.empty) = (excludes ++ extra).exists(t <:< _)
   }
 
   implicit class ReflectionHelpers(t: Type) {

--- a/swagger/src/test/scala/org/http4s/rho/swagger/SwaggerModelsBuilderSpec.scala
+++ b/swagger/src/test/scala/org/http4s/rho/swagger/SwaggerModelsBuilderSpec.scala
@@ -13,6 +13,8 @@ import scodec.bits.ByteVector
 import scala.reflect.runtime.universe._
 
 import scalaz._, Scalaz._
+import scalaz.stream._
+import scalaz.concurrent.Task
 
 class SwaggerModelsBuilderSpec extends Specification {
   import models._, DummyCompiler.compilerInstance
@@ -286,6 +288,48 @@ class SwaggerModelsBuilderSpec extends Specification {
 
     "collect response of a collection of tuples" in {
       val ra = GET / "test" |>> { () => Ok(List((0, ModelA("", 0)))) }
+
+      sb.collectResponses(ra) must havePair(
+        "200" -> Response(
+          description = "OK",
+          schema      = ArrayProperty(items = RefProperty(ref = "Tuple2«Int,ModelA»")).some))
+    }
+
+    "collect response of a Process of primitives" in {
+      val ra1 = GET / "test" |>> { () => Ok(Process.eval(Task(""))) }
+      val ra2 = GET / "test" |>> { () => Ok(Process.emit("")) }
+
+      sb.collectResponses(ra1) must havePair(
+        "200" -> Response(description = "OK", schema = AbstractProperty(`type` = "string").some))
+
+      sb.collectResponses(ra2) must havePair(
+        "200" -> Response(description = "OK", schema = AbstractProperty(`type` = "string").some))
+    }
+
+    "collect response of a Process of non-primitives" in {
+      val ra1 = GET / "test" |>> { () => Ok(Process.eval(Task(List((0, ModelA("", 0)))))) }
+      val ra2 = GET / "test" |>> { () => Ok(Process.emit(List((0, ModelA("", 0))))) }
+
+      sb.collectResponses(ra1) must havePair(
+        "200" -> Response(
+          description = "OK",
+          schema      = ArrayProperty(items = RefProperty(ref = "Tuple2«Int,ModelA»")).some))
+
+      sb.collectResponses(ra2) must havePair(
+        "200" -> Response(
+          description = "OK",
+          schema      = ArrayProperty(items = RefProperty(ref = "Tuple2«Int,ModelA»")).some))
+    }
+
+    "collect response of a Task of a primitive" in {
+      val ra = GET / "test" |>> { () => Ok(Task("")) }
+
+      sb.collectResponses(ra) must havePair(
+        "200" -> Response(description = "OK", schema = AbstractProperty(`type` = "string").some))
+    }
+
+    "collect response of a Task of a non-primitive" in {
+      val ra = GET / "test" |>> { () => Ok(Task(List((0, ModelA("", 0))))) }
 
       sb.collectResponses(ra) must havePair(
         "200" -> Response(

